library/src/main/scala/scala/scalajs/runtime/RuntimeLong.scala

```diff
@@ -302,9 +302,212 @@ final class RuntimeLong(val lo: Int, val hi: Int)
 
   @inline
   def *(b: RuntimeLong): RuntimeLong = {
+    /* The following algorithm is based on the decomposition in 32-bit and then
+     * 16-bit subproducts of the unsigned interpretation of operands.
+     *
+     * We are looking to compute
+     * a *[64] b = (a * b) % 2^64
+     *
+     * We use the notation * and + for mathematical, non-overflowing
+     * operations, and *[64] and +[64] for overflowing operations. Eventually,
+     * we need to implement everything in terms of *[32] and +[32]. The symbol
+     * ^ is used for exponentiation (nor bitwise xor).
+     *
+     * The decomposition in 32-bit components yields:
+     *
+     * a *[64] b
+     *   = ( (2^32*ahi + alo) * (2^32*bhi + blo) ) % 2^64
+     *   = ( 2^64*ahi*bhi + 2^32*(ahi*blo + bhi*alo) + alo*blo ) % 2^64
+     *
+     * With Natural numbers, congruence theory tells us that we can "distribute"
+     * `% n` on + and *, as long as we also keep the other `% n`. To be more
+     * precise:
+     *   (a + b) % n = (a%n + b) % n = (a + b%n) % n = (a%n + b%n) % n
+     *   (a * b) % n = (a%n * b) % n = (a * b%n) % n = (a%n * b%n) % n
+     *
+     * We can use these equivalence to get rid of parts of our computation:
```

Comment:
There are two of them just above. (although I'm not sure I actually use the second one; I'll check)